autotst/species.py

```diff
@@ -502,6 +519,44 @@ def get_torsions(self):
 
                 torsion_tup = (atom0.GetIdx(), atom1.GetIdx(),
                                atom2.GetIdx(), atom3.GetIdx())
+                description = 'hindered'
+
+                endpoints = check_angles(torsion_tup)
+                center_atoms = []
+   
+                while len(endpoints) > 0:
```

Comment:
Please add comments here, it's hard to understand what's going on here.

Reply:
I'm not sure if I'm tired cuz it's late, but this is a bit confusing. Could you explain this section of the code?

Reply:
I added some comments, although it is very confusing.  Maybe we could try to think of a cleaner algorithm?